volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,151 @@
+                    
+                    # Standard Information Attribute
+                    if attr_header.AttrType == 0x10:
+                        attr_data = self.context.object(si_object, offset=attr_data_offset, layer_name=layer.name)
+
+                        yield 0, (
+                            format_hints.Hex(attr_data_offset),
+                            mft_record.get_signature(),
+                            mft_record.RecordNumber,
+                            mft_record.LinkCount,
+                            MFTFlags(mft_record.Flags).name,
+                            renderers.NotApplicableValue(),
+                            AttributeTypes(attr_header.AttrType).name,
+                            conversion.wintime_to_datetime(attr_data.CreationTime),
+                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
+                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
+                            conversion.wintime_to_datetime(attr_data.AccessedTime),
+                            renderers.NotApplicableValue(),
+                        )
+
+                    # File Name Attribute
+                    if attr_header.AttrType == 0x30:
+                        attr_data = self.context.object(fn_object, offset=attr_data_offset, layer_name=layer.name)
+                        file_name = attr_data.get_full_name()
+
+                        yield 1, (
+                            format_hints.Hex(attr_data_offset),
+                            mft_record.get_signature(),
+                            mft_record.RecordNumber,
+                            mft_record.LinkCount,
+                            MFTFlags(mft_record.Flags).name,
+                            PermissionFlags(attr_data.Flags).name,
+                            AttributeTypes(attr_header.AttrType).name,
+                            conversion.wintime_to_datetime(attr_data.CreationTime),
+                            conversion.wintime_to_datetime(attr_data.ModifiedTime),
+                            conversion.wintime_to_datetime(attr_data.UpdatedTime),
+                            conversion.wintime_to_datetime(attr_data.AccessedTime),
+                            file_name
+                        )
+                    
+                    # Update the base offset to point to the next attribute
+                    attr_base_offset += attr_header.Length
+
+            except exceptions.PagedInvalidAddressException:
+                pass
+
+    def generate_timeline(self):
+        for row in self._generator():
+            if row[-1] != 'N/A':
+                filename = row[-1]
+                created = f'File {row[-1]} Created'
+                updated = f'File {row[-1]} Updated'
+                modified = f'File {row[-1]} Modified'
+                accessed = f'File {row[-1]} Accessed'
+                yield (f'File {filename} created', timeliner.TimeLinerType.CREATED, row[7])
```

Comment:
I think it would be more useful to return these all in one go, since they relate to the same item.  Since these are the same file, the item value should match (probably just the filename).  The timeliner plugin will then be able to put them on the same row and sort/duplicate them as necessary.  If the documentation needs updating to make this clearer just lemme know and I'll tart it up appropriately...  5:)

Reply:
Took me a while to realize the description / info is part of the key to put all the times on to a single row.